### PR TITLE
NAS-108931 / 21.02 / Set ZFS_INITRD_POST_MODPROBE_SLEEP if boot pool is on USB

### PR DESCRIPTION
**Pull request by themylogin**

### CODE REVIEW

File: src/freenas/usr/local/bin/initrd-zfs.py [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+            mapping[dev.sys_name] = dev.get("ID_BUS")
+        elif dev.get("ID_PART_ENTRY_UUID"):
+            parent = dev.find_parent("block")
+            mapping[dev.sys_name] = parent.get("ID_BUS")
+            mapping[os.path.join("disk/by-partuuid", dev.get("ID_PART_ENTRY_UUID"))] = parent.get("ID_BUS")
+
+    has_usb = False
+    for dev in disks:
+        if mapping.get(dev) == "usb":
+            has_usb = True
```
Comment (william-gr) [inline src/freenas/usr/local/bin/initrd-zfs.py:31]: break?

File: src/freenas/usr/local/bin/initrd-zfs.py [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    zfs_var_name = "ZFS_INITRD_POST_MODPROBE_SLEEP"
+    lines = [line for line in lines if not line.startswith(f"{zfs_var_name}=")]
+    if has_usb:
+        lines.append(f"{zfs_var_name}=15")
+
+    with open(zfs_config_path, "w") as f:
+        new_config = "\n".join(lines) + "\n"
+        f.write(new_config)
+
+    if update_initramfs_if_changes == "1" and new_config != original_config:
```
Comment (william-gr) [inline src/freenas/usr/local/bin/initrd-zfs.py:47]: Should we write zfs_config_path only here? so we dont write the config if nothing has changed?

File: src/middlewared/middlewared/plugins/boot.py [hunk 3]
```diff
@@ -172,6 +176,11 @@ async def get_scrub_interval(self):
         """
         return (await self.middleware.call('system.advanced.config'))['boot_scrub']
 
+    @private
+    async def handle_initrd_zfs(self, update_initramfs_if_changes=True):
+        await run('/usr/local/bin/initrd-zfs.py', BOOT_POOL_NAME, '/', str(int(update_initramfs_if_changes)),
+                  encoding='utf8', errors='ignore')
```
Comment (william-gr) [inline src/middlewared/middlewared/plugins/boot.py:182]: Seems like we could capture output and log if anything wrong happens with that script?

**Comment by william-gr** [comment]
@themylogin ping

File: src/middlewared/middlewared/plugins/boot.py (hunk 3 above)
Comment (themylogin) [reply src/middlewared/middlewared/plugins/boot.py:182]: This is done automatically, unhandled `subprocess.CalledProcessError` is re-raised as CallError with failed command's output

**Review by william-gr — APPROVED** [review]

**Pull request merged**